File: projet_groupe4/src/main/java/projet_groupe4/model/Badge.java
package projet_groupe4.model;



import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;


@Entity
@Table(name="badge")
public class Badge {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	
	private Integer id;
	@Column(name="nom_badge", columnDefinition="varchar(30)", nullable=false)
	
	private String nomBadge;
	@Column(name="point_min", nullable=false)
	
	private int pointMin;
	@Column(name="img_url")
	
	private String imgURL;
	
	// Constructors
	public Badge() {}
	public Badge(Integer id, String nomBadge, int pointMin, String imgURL) {
		this.id = id;
		this.nomBadge = nomBadge;
		this.pointMin = pointMin;
		this.imgURL = imgURL;
	}
	public Badge(String nomBadge, int pointMin, String imgURL) {
		this.nomBadge = nomBadge;
		this.pointMin = pointMin;
		this.imgURL = imgURL;
	}

	// Getters & Setters
	public Integer getId() {
		return id;
	}
	public void setId(Integer id) {
		this.id = id;
	}
	public String getNomBadge() {
		return nomBadge;
	}
	public void setNomBadge(String nomBadge) {
		this.nomBadge = nomBadge;
	}
	public int getPointMin() {
		return pointMin;
	}
	public void setPointMin(int pointMin) {
		this.pointMin = pointMin;
	}
	public String getImgURL() {
		return imgURL;
	}
	public void setImgURL(String imgURL) {
		this.imgURL = imgURL;
	}
	
	// toString
	@Override
	public String toString() {
		return "Badge [id=" + id + ", nomBadge=" + nomBadge + ", pointMin=" + pointMin + ", imgURL=" + imgURL + "]";
	}

}

File: projet_groupe4_angular/src/app/dto/badge.ts
export class Badge {

    constructor(
        public _id: number,
        public _nomBadge: string,
        public _pointMin: number,
        public _imgURL: string
    ) {}

    public get id(): number {
        return this._id;
    }

    public set id(value: number) {
        this._id = value;
    }

    public get nomBadge(): string {
        return this._nomBadge;
    }

    public set nomBadge(value: string) {
        this._nomBadge = value;
    }

    public get pointMin(): number {
        return this._pointMin;
    }

    public set pointMin(value: number) {
        this._pointMin = value;
    }

    public get imgURL(): string {
        return this._imgURL;
    }

    public set imgURL(value: string) {
        this._imgURL = value;
    }

    public toJson(): any {
        return {
            nomBadge: this._nomBadge,
            pointMin: this._pointMin,
            imgURL: this._imgURL
        };
    }
}
